components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Extensions/DataSetWriterModelEx.cs

```diff
@@ -40,11 +40,11 @@ public static class DataSetWriterModelEx {
             }
             var model = new SubscriptionModel {
                 Connection = dataSetWriter.DataSet.DataSetSource.Connection.Clone(),
-                Id = dataSetWriter.DataSetWriterId,
+                Id = dataSetWriter.DataSetWriterName,
                 MonitoredItems = monitoredItems,
                 ExtensionFields = dataSetWriter.DataSet.ExtensionFields,
                 Configuration = dataSetWriter.DataSet.DataSetSource
-                    .ToSubscriptionConfigurationModel()
+                    .ToSubscriptionConfigurationModel(dataSetWriter.DataSet.DataSetMetaData)
```

Comment:
do we need to check if DataSet is null here?

Reply:
Good one. I will add a Debug.Assert or  null check.

Reply:
Checked on line 27.